### PR TITLE
Correctly setup compaction sources in block meta files

### PR DESCRIPTION
**Pull request by cyriltovena**
The original block should have all source IDS which is itself, then compacted should merge source of parents.

Also adds test for both case.

### CODE REVIEW

File: pkg/phlaredb/head.go [hunk 1]
```diff
@@ -1003,6 +1004,7 @@ func (h *Head) flush(ctx context.Context) error {
 	h.meta.Files = files
 	h.meta.Stats.NumProfiles = uint64(h.profiles.index.totalProfiles.Load())
 	h.meta.Stats.NumSamples = h.totalSamples.Load()
+	h.meta.Compaction.Sources = []ulid.ULID{h.meta.ULID}
```
Comment (cyriltovena) [inline pkg/phlaredb/head.go:1007]: Original block should have their own sources this is how the Prometheus format expect it.

Comment (cyriltovena) [reply pkg/phlaredb/head.go:1007]: cc @pracucci if we don't have local compaction that is correct right ?

Comment (cyriltovena) [reply pkg/phlaredb/head.go:1007]: Confirmed this with Marco and reading more Prometheus.

**Review by kolesnikovae — APPROVED** [review]
LGTM

File: pkg/phlaredb/head.go (hunk 1 above)
Comment (simonswine) [inline pkg/phlaredb/head.go:1007]: When prometheus does this it also sets compaction level to `1`

Comment (simonswine) [reply pkg/phlaredb/head.go:1007]: https://github.com/prometheus/prometheus/blob/76dd9b5470e6f49330a6420c5d6133bea40a71ea/tsdb/compact.go#L499C1-L500

**Pull request merged**